### PR TITLE
refactor(custom-connector): POC cleanup

### PR DESCRIPTION
**Pull request by zregvart**
This cleans up implementation and tests of the Swager custom connector
generator.

### CODE REVIEW

File: rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import io.swagger.models.Swagger;
+import io.syndesis.core.Json;
+
+public final class SwaggerHelper {
+
+    private SwaggerHelper() {
+        // utility class
+    }
+
+    /* default */ static String serialize(final Swagger swagger) {
```
Comment (rhuss) [inline rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java]: Is this helper class really needed ? The method is only called once and hence could be inlined, too.

File: rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java [hunk 2]
```diff
@@ -48,67 +44,77 @@
...
     }
 
     @Override
     public Connector info(final ConnectorTemplate connectorTemplate, final ConnectorSettings connectorSettings) {
         return basicConnector(connectorTemplate, connectorSettings);
     }
 
-    /* default */ Connector basicConnector(final ConnectorTemplate connectorTemplate,
-        final ConnectorSettings connectorSettings) {
+    /* default */ Connector basicConnector(final ConnectorTemplate connectorTemplate, final ConnectorSettings connectorSettings) {
```
Comment (rhuss) [inline rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java]: any particular reason why these methods are not `private` ? For testing ?

File: rest/connector-generator/src/main/java/io/syndesis/connector/generator/ConnectorGenerator.java [hunk 3]
```diff
@@ -44,10 +43,13 @@ protected final Connector baseConnectorFrom(final ConnectorTemplate connectorTem
         final String name = Optional.ofNullable(connectorSettings.getName())
             .orElseGet(() -> determineConnectorName(connectorTemplate, connectorSettings));
 
+        final String description = Optional.ofNullable(connectorSettings.getDescription())
+            .orElseGet(() -> determineConnectorDescription(connectorTemplate, connectorSettings));
+
```
[the remark this answers is not in the document]
Comment (zregvart) [reply rest/connector-generator/src/main/java/io/syndesis/connector/generator/ConnectorGenerator.java]: thanks 💯 -- must be the hipster in me, I'll create a PR to clean this up a bit.

File: rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java (hunk 2 above)
Comment (zregvart) [reply rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerConnectorGenerator.java]: I can't see it being used in tests, I guess I intended to. I usually prefer package private methods to private, that could also be a hipster thing...

File: rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java (hunk 1 above)
Comment (zregvart) [reply rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java]: I refactored it out seemed at the point that I might be used more. I was getting (correct) complaints from PMD about SwaggerConnectorGenerator being a god class...

**Pull request merged**